tests/hypervisor/test_esx.py

```diff
@@ -93,6 +93,226 @@ def test_hypervisor_id(self, virtwho, function_hypervisor, hypervisor_data, glob
                     assert not satellite.host_id(hypervisor_data['hypervisor_hostname'])
                     assert not satellite.host_id(hypervisor_data['hypervisor_uuid'])
 
+    @pytest.mark.tier1
+    def test_filter_and_exclude_hosts(self, virtwho, function_hypervisor, hypervisor_data):
+        """Test the filter_hosts= and exclude_hosts= option in /etc/virt-who.d/hypervisor.conf
+
+        :title: virt-who: esx: test filter_hosts and exclude_hosts function
+        :id: 3c4f3f42-a10c-4205-baa9-7464e3f6870a
+        :caseimportance: High
+        :tags: tier1
+        :customerscenario: false
+        :upstream: no
+        :steps:
+            1. Set hypervisor_id=host_uuid.
+            2. Configure filter_hosts=host_uuid, run the virt-who service.
+            3. Configure filter_hosts='*', run the virt-who service.
+            4. Configure filter_hosts='', run the virt-who service.
+            5. Delete filter_hosts, Configure exclude_hosts=host_uuid, run the virt-who service.
+            6. Configure exclude_hosts='*', run the virt-who service.
+            7. Configure exclude_hosts='', run the virt-who service.
+            8. run the above cases with hypervisor_id =hostname and hwuuid
+            9. run virt-who with filter_hosts=[host_uuid] and exclude_hosts=[host_uuid]
+            10. run virt-who with filter_hosts=* and exclude_hosts=[host_uuid]
+            11. run virt-who with exclude_hosts= and filter_hosts=[host_uuid]
+
+        :expectedresults:
+            2. Succeeded to run the virt-who, can find the log message "hypervisorId": "{host_uuid}"
+            3. Succeeded to run the virt-who, can find the log message "hypervisorId": "{host_uuid}"
+            4. Succeeded to run the virt-who, cannot find the log message "hypervisorId": "{host_uuid}"
+            5. Succeeded to run the virt-who, cannot find the log message "hypervisorId": "{host_uuid}"
+            6. Succeeded to run the virt-who, cannot find the log message "hypervisorId": "{host_uuid}"
+            7. Succeeded to run the virt-who, can find the log message "hypervisorId": "{host_uuid}"
+            8. The result is the same as above
+            9. Succeeded to run the virt-who, cannot find the log message "hypervisorId": "{host_uuid}"
+            10. Succeeded to run the virt-who, cannot find the log message "hypervisorId": "{host_uuid}"
+            11. Succeeded to run the virt-who, can find the log message "hypervisorId": "{host_uuid}"
+
+        """
+        hypervisor_ids = ['hostname', 'uuid', 'hwuuid']
+        for hypervisor_id in hypervisor_ids:
+            function_hypervisor.update('hypervisor_id', hypervisor_id)
+            hypervisor_id_data = hypervisor_data[f'hypervisor_{hypervisor_id}']
+
+            # run virt-who with filter_hosts option
+            for filter_hosts in [hypervisor_id_data, '*', '']:
```

Comment:
I think we can just test the `hypervisor_id_data` in this step, because the `* and null` value also covered by the tier2 case, and we should keep the tier1 cases clear and only focused on the basic function.